src/Analyser/NodeScopeResolver.php

```diff
@@ -593,12 +593,25 @@ private function processStmtNode(
 
 					$gatheredReturnStatements[] = new ReturnStatement($scope, $node);
 				}, StatementContext::createTopLevel());
+
+				$classReflection = $methodScope->getClassReflection();
```

Comment:
You could do this on `$scope` and then you don't need to check for `null` - because of `if (!$scope->isInClass()) {` on line 526.

Reply:
is the `isInClass` thing a custom type-specifier in phpstan?

Reply:
Not anymore, it just has `@phpstan-assert` above it.

Reply:
ah i didn't know they could do that, nice